lua/entities/gmod_artron_inhibitor/init.lua

```diff
@@ -0,0 +1,214 @@
+-- Artron Inhibitor by Nova Astral, an edit of the Time Distortion Generator by parar020100 and JEREDEK
+
+AddCSLuaFile("cl_init.lua")
+AddCSLuaFile("shared.lua")
+
+include("shared.lua")
+
+
+function ENT:Initialize()
+    self:SetModel("models/props_lab/reciever01b.mdl")
+    self:PhysicsInit(SOLID_VPHYSICS)
+    self:SetMoveType(MOVETYPE_VPHYSICS)
+    self:SetSolid(SOLID_VPHYSICS)
+
+    self:SetName("ArtronInhibitor")
+    self:SetColor(Color(255, 255, 255, 255))
+    self:SetUseType(SIMPLE_USE)
+
+    local phys = self:GetPhysicsObject()
+
+    self.Radius = 1000
+    self.EntMaxHealth = 100
+    self.EntHealth = self.EntMaxHealth
+    
+
+    if phys:IsValid() then
+        phys:Wake()
+    end
+
+    if WireLib then
+        local inNames = {"Activate","Radius"}
+        local inTypes = {"NORMAL","NORMAL"}
+
+        WireLib.CreateSpecialInputs( self,inNames,inTypes)
+        Wire_CreateOutputs(self,{"Active","Radius","Health"})
+        
+        Wire_TriggerOutput(self,"Radius",self.Radius)
+        Wire_TriggerOutput(self,"Health",self.EntHealth)
+    end
+end
+
+function ENT:TriggerWire(tname,value)
+    if WireLib then
+        Wire_TriggerOutput(self,tname,value)
+    end
+end
+
+function ENT:GetRadius()
+    return self.Radius
+end
+
+function ENT:SetRadius(radius)
+    self.Radius = math.Clamp(radius,0,2048)
+    self:TriggerWire("Radius",self.Radius)
+    return self.Radius
+end
+
+function ENT:Repair(repair)
```

Comment:
We would also want to add this into the sonic screwdriver so it can be repaired etc, can you make another PR for that?

I've added you as a collaborator to that repo now

Reply:
made a pr in the sonicsd for this